promql/info_test.go

```diff
@@ -0,0 +1,492 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package promql_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/promqltest"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+func TestInfoFunc(t *testing.T) {
```

Comment:
I played around with converting these test cases into promqltest acceptance tests and it highlighted an issue with step invariant evaluation of the info function. I think these should go into promqltest acceptance tests since they cover more evaluations.

Reply:
Sketched it here: https://github.com/MichaHoffmann/prometheus/commit/a13bb2ccd50c5bc7bb35a8773030e3943ce1ac36